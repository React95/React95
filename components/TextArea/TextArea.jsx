import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Text = styled.textarea`
  border: none;
  border-left: 1px solid #868a8e;
  border-top: 1px solid #868a8e;

  font-size: 14px;
  line-height: 18px;

  outline: none;
  resize: none;

  box-shadow: inset -1px -1px 0 0 #c3c7cb, inset 1px 1px 0 0 #000000,
    0.5px 0.5px 0 0.5px #ffffff;
`;

const TextArea = ({ rows, cols, ...rest }) => (
  <Text {...rest} className="text" rows={rows} cols={cols} />
);

TextArea.propTypes = {
  rows: PropTypes.number,
  cols: PropTypes.number,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

TextArea.defaultProps = {
  rows: 10,
  cols: 50,
  children: null,
};

export default TextArea;
