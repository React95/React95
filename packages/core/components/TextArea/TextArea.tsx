import * as React from 'react';
import { Frame, FrameProps } from '../Frame/Frame';
import { input as textarea } from '../Input/Input.css';

export type TextAreaProps = Omit<FrameProps<'textarea'>, 'as'>;

export const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
  (rest, ref) => (
    <Frame {...rest} ref={ref} className={textarea} as="textarea" />
  ),
);
