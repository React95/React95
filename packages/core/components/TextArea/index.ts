import { TextArea } from './TextArea';

export default TextArea;
