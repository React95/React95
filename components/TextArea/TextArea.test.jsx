import React from 'react';
import { render } from 'react-testing-library';
import TextArea from './TextArea';

describe('<TextArea />', () => {
  describe('Snapshot', () => {
    it('should match snapshot', () => {
      const { container } = render(<TextArea />);
      expect(container).toMatchSnapshot();
    });
  });
});
