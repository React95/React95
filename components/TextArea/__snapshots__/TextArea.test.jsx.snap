// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> Snapshot should match snapshot 1`] = `
.c0 {
  border: none;
  border-left: 1px solid #868a8e;
  border-top: 1px solid #868a8e;
  font-size: 14px;
  line-height: 18px;
  outline: none;
  resize: none;
  box-shadow: inset -1px -1px 0 0 #c3c7cb,inset 1px 1px 0 0 #000000,0.5px 0.5px 0 0.5px #ffffff;
}

<div>
  <textarea
    class="c0"
    cols="50"
    rows="10"
  />
</div>
`;
